### ISSUE TITLE
CCTM breaks Simple Page Ordering functionality

### ISSUE THREAD

**Issue by GoogleCodeExporter** [viktorthang/wordpress-custom-content-type-manager#180]
```
What steps will reproduce the problem? (Be specific)
1. Install Simple Page Ordering (SPOR)
(http://wordpress.org/extend/plugins/simple-page-ordering/)
Ordering pages via plugin drag and drop UI works.
2. Install CCTM
3. Ordering pages or custom content types with "Page attributes - Hierarchical" 
via plugin drag and drop UI does not work.

The UI works, however results are not saved.

What is the expected output? What do you see instead?
Retaining Simple Page Ordering functionality, being able to order pages and 
custom content types with the drag and drop interface.

On the SPOR homepage it says his plugin does in fact support custom content 
types. And from the comments other people seem to have made it work, presumably 
with coding their custom content types or using other plugins.

Does the problem continue if you disable all other plugins?
If you disable CCTM, the SPOR functionality returns.

I went over the SPOR code, and it fails silently at line 137 with 
wp_update_post. As a matter of fact, any call to wp_update_post within that 
plugin fails with no error, but -1 returned by wp_ajax as long as CCTM is 
active.


Please provide any additional information below (e.g. Browser, version,
OS):
WP 3.2.1
CCTM 0.9.3.3 (From WP site, not dev version)
Simple Page Ordering: 1.0
PHP Version: 5.3.5

```

Original issue reported on code.google.com by `Oph...@gmail.com` on 19 Sep 2011 at 2:57

**Comment by GoogleCodeExporter** [comment]
```
Thanks for taking the time to file a detailed bug report!

Initially, this sounds similar sounding to issue 169, but I bet it's actually 
the issue that was outlined in issue 52 (i.e. problems with wp_insert_post).  
Try editing the 
wp-content/plugins/custom-content-type-manager/includes/StandardizedCustomFields
.php file... line 286 contains this:
        if ( !empty($_POST) && check_admin_referer('update_custom_content_fields','custom_content_fields_nonce') ) {

Try editing it so it reads like this:
        if ( !empty($_POST)) {


In other words, remove the check_admin_referer() from that if statement, save 
the file, and see if that doesn't clear up this problem.  This issue has been 
corrected and will be available in the next release (0.9.4).  Please let me 
know if that fixes the problem.

```

Original comment by `fireproofsocks` on 19 Sep 2011 at 5:29
- Changed state: **InReview**

**Comment by GoogleCodeExporter** [comment]
```
I can confirm that once I apply your fix the ajax doesn't die with -1

However - all custom fields for the changed posts get deleted :( 
So it's a no go
```

Original comment by `Oph...@gmail.com` on 19 Sep 2011 at 9:42

**Comment by GoogleCodeExporter** [comment]
```
Hmm...  I have to think about that one for a bit.  I think what's going on is 
that both plugins are hooking into the "save_post" action, and they aren't 
playing nicely with each other... WP just doesn't have a good architecture for 
this, so it's a bit frustrating.  What I *think* is happening is since both 
plugins are tied to that event, I have to figure out a way to determine if the 
action is being triggered by the normal WP create/update post procedure, or if 
it is being triggered from the front end (as in issue 52), or from some other 
plugin (as in this case).

So the fix for issue 52 was to avoid checking the admin referrer and just check 
to see if there was $_POST data before doing the saving of custom fields.  But 
if SPOR is sending data via $_POST, then the CCTM saving of custom fields will 
still be triggered, and since the custom fields aren't actually being saved, it 
interprets them as being empty, so it deletes all their values.

Here's my fix:

Inside of includes/StandardizedCustomFields.php

inside the print_custom_fields() function around line 220, put in this:

        // Print the nonce: this offers security and it will help us know when we should do custom saving logic in the save_custom_fields function
        $output .= '<input type="hidden" name="_cctm_nonce" value="'. wp_create_nonce('cctm_create_update_post') . '" />';

Put it right above this stuff:

        // Print the form
        print '<div class="form-wrap">';        
        print $output;
        print '</div>';


And then, later in that file inside the save_custom_fields() function, put in 
this:

        // Use this to ensure you save custom fields only when saving from the edit/create post page
        $nonce = CCTM::get_value($_POST, '_cctm_nonce');
        if (! wp_verify_nonce($nonce, 'cctm_create_update_post') ) {
            return;
        }

Right before this:
        if ( !empty($_POST) ) {         

That should do it.  That should make the CCTM handling kick in only when you 
create or edit a post from inside the manager.  I've not been able to verify 
the original behavior, though...  even without this fix, I couldn't make my 
custom fields get destroyed.  If the above doesn't fix it, could you elaborate 
a bit more on when exactly the custom fields get nuked?  Thanks.
```

Original comment by `fireproofsocks` on 19 Sep 2011 at 10:59

**Comment by GoogleCodeExporter** [comment]
```
You had a typo there. It should be CCTM::_get_value instead of CCTM::get_value

I applied your fix and everything works perfectly!
Bug squashed :)
```

Original comment by `Oph...@gmail.com` on 20 Sep 2011 at 9:26

**Comment by GoogleCodeExporter** [comment]
```
Ah, typos... the bane of my existence.  Thanks for pointing that out!  Glad 
this works.  It's been committed for version 0.9.4. 
```

Original comment by `fireproofsocks` on 20 Sep 2011 at 2:22
- Changed state: **Fixed**
